src/overlays/actors/ovl_En_Encount1/z_en_encount1.c

```diff
@@ -19,13 +27,310 @@ const ActorInit En_Encount1_InitVars = {
+        if ((floorType != 4) && (floorType != 7) && (floorType != 12)) {
+            this->numLeeverSpawns = 0;
+        } else if (!(this->reduceLeevers && (this->actor.xzDistFromLink > 1300.0f))) {
+            spawnLimit = 5;
+            if (this->reduceLeevers) {
+                spawnLimit = 3;
+            }
+            while ((this->curNumSpawn < this->maxCurSpawns) && (this->curNumSpawn < spawnLimit) && (this->timer == 0)) {
+                spawnDist = sLeeverDist[this->leeverIndex];
+                spawnAngle = sLeeverAngle[this->leeverIndex] + player->actor.shape.rot.y;
+                spawnParam = LEEVER_SMALL;
+
+                if ((this->killCount >= 10) && (this->bigLeever == NULL)) {
+                    this->killCount = this->numLeeverSpawns = 0;
+                    spawnAngle = sLeeverAngle[0];
+                    spawnDist = sLeeverDist[2];
+                    spawnParam = LEEVER_BIG;
+                }
+
+                spawnPos.x = player->actor.posRot.pos.x + Math_Sins(spawnAngle) * spawnDist;
+                spawnPos.y = player->actor.groundY + 120.0f;
+                spawnPos.z = player->actor.posRot.pos.z + Math_Coss(spawnAngle) * spawnDist;
+
+                floorY = func_8003C9A4(&globalCtx->colCtx, &floorPoly, &bgId, &this->actor, &spawnPos);
+                if (floorY <= -32000.0f) {
+                    return;
+                }
+                spawnPos.y = floorY;
+
+                leever = (EnReeba*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_REEBA,
+                                                      spawnPos.x, spawnPos.y, spawnPos.z, 0, 0, 0, spawnParam);
+                if (1) {}
+                if (1) {}
+                if (leever != NULL) {
+                    this->curNumSpawn++;
+                    leever->unk_280 = this->leeverIndex++;
+                    if (this->leeverIndex >= 5) {
+                        this->leeverIndex = 0;
+                    }
+                    this->numLeeverSpawns++;
+                    if (this->numLeeverSpawns >= 12) {
+                        this->timer = 150;
+                        this->numLeeverSpawns = 0;
+                    }
+                    if (spawnParam != LEEVER_SMALL) {
+                        this->timer = 300;
+                        this->bigLeever = leever;
+                    }
+                    if (!this->reduceLeevers) {
+                        this->maxCurSpawns = (s16)Math_Rand_ZeroFloat(3.99f) + 2;
+                    } else {
+                        this->maxCurSpawns = (s16)Math_Rand_ZeroFloat(2.99f) + 1;
+                    }
+                } else {
+                    // Cannot spawn!
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+                    return;
```

Comment:
That's non equivalent because it's inside a while loop. However, all of the early returns in the while loops could instead be `break`. Is that better?

Reply:
I think so yeah